tools/DfmHttpService/DfmHttpService.cs

```diff
@@ -0,0 +1,168 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace DfmHttpService
+{
+    using System;
+    using System.IO;
+    using System.Net;
+    using System.Text;
+
+    using Microsoft.DocAsCode.Build.Engine;
+    using Microsoft.DocAsCode.Plugins;
+
+    using Newtonsoft.Json;
+
+    public class DfmHttpService
+    {
+        private bool KeepGoing { get; set; } = true;
+        private const string UrlPrefix = "http://localhost:4001/";
+
+        public void StartService(string urlPrefix)
+        {
+            var listener = new HttpListener();
+            listener.Prefixes.Add(urlPrefix);
+            listener.Start();
+
+            while (KeepGoing)
+            {
+                var context = listener.GetContext();
+                HandleRequest(context);
+            }
+
+            listener.Stop();
+            listener.Close();
+        }
+
+        private void HandleRequest(HttpListenerContext context)
+        {
+            // TODO: Add log for request information
+            var request = context.Request;
+
+            CommandMessage command;
+            try
+            {
+                command = GetCommandMessage(request);
+            }
+            catch (Exception ex)
+            {
+                ReplyClientErrorResponse(context, ex.Message);
+                return;
+            }
+
+            switch (command.Name)
+            {
+                case "preview":
```

Comment:
I think it's clear to make the command name in request body decalre as string instead of interger. Maybe I can wrap it into a constant class...

---
In reply to: [94093188](https://github.com/dotnet/docfx/pull/1075#discussion_r94093188) [](ancestors = 94093188)